### PR TITLE
[CARBONDATA-2705][CarbonStore] CarbonStore Java API and Implementation

### PR DESCRIPTION
**Pull request by jackylk**
Support two implementations:
1. LocalCarbonStore for usage in local mode
2. DistributedCarbonStore leveraging multiple server (Master and Workers) via RPC

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6884/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6889/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5666/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6904/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5670/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5686/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6906/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5672/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5673/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5691/

File: format/src/main/thrift/schema.thrift [hunk 1]
```diff
@@ -196,7 +196,7 @@ struct DataMapSchema  {
     // class name
     2: required string className;
     // to maintain properties which are mentioned in DMPROPERTIES of DDL and also it
-    // stores properties of select query, query type like groupby, join in
```
Comment (QiangCai) [inline format/src/main/thrift/schema.thrift]: not required this modification.
maybe it is a re-factory mistake.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/detailquery/SearchModeTestCase.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ import org.apache.carbondata.spark.util.DataGenerator
 /**
  * Test Suite for search mode
  */
-
+@Ignore
```
Comment (QiangCai) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/detailquery/SearchModeTestCase.scala]: why do you ignore these test cases?

**Pull request closed without merging**